client/image_save.go

```diff
@@ -2,17 +2,33 @@ package client // import "github.com/docker/docker/client"
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
 	"io"
 	"net/url"
+
+	"github.com/docker/docker/api/types/image"
 )
 
 // ImageSave retrieves one or more images from the docker host as an io.ReadCloser.
 // It's up to the caller to store the images and close the stream.
-func (cli *Client) ImageSave(ctx context.Context, imageIDs []string) (io.ReadCloser, error) {
+func (cli *Client) ImageSave(ctx context.Context, imageIDs []string, opts image.SaveOptions) (io.ReadCloser, error) {
 	query := url.Values{
 		"names": imageIDs,
 	}
 
+	if opts.Platform != nil {
+		if err := cli.NewVersionError(ctx, "1.47", "platform"); err != nil {
+			return nil, err
+		}
+
+		p, err := json.Marshal(*opts.Platform)
+		if err != nil {
+			return nil, fmt.Errorf("invalid platform: %v", err)
```

Comment:
Same here for considering to use an `errdefs`.


☝️ Actually starting to wonder if it would make sense to have a utility for this; yes, it would be minimal (just json marshalling and handling errors), but would help producing the right errors and being consistent WDYT?


☝️ And I just realise we have `httputils.DecodePlatform`, so to some extent it would be nice if we had the reverse to that, but I guess `httputils` is part of `api/server`, so if we do, we should look where a suitable location would be if we want to put them together 🤔